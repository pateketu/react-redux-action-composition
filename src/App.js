import React, { Component } from 'react';
import Chat from './component/chat'
import './App.css';
class App extends Component {
  render() {
    return (      
        <Chat/>
    );
  }
}

export default App;
