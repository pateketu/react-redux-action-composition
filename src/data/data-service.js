import {getMessages} from './data';
class DataService{
    fetchMessages(){
        return getMessages();
    }
}
export default  new DataService();