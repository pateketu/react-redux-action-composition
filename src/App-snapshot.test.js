import React from 'react';
import App from './App';
import store from './redux/store';
import {Provider} from 'react-redux'
import {mount} from 'enzyme';
import {getMessages, getMembers} from './data/data';
import toJson from 'enzyme-to-json';

jest.mock('./data/data');

describe('App', ()=>{
    
    const messagePromise = getMessages.mockImplementation(()=>{
        return Promise.resolve([{
            "id": "cd445e6d-e514-424f-ba8f-16ec842002c6",
            "userId": "e837c9f5-247f-445f-bcc3-7d434348336b",
            "message": "Test Message",
            "timestamp": "2017-02-09T04:27:38Z"
          }]);
    });

    const memberPromise = getMembers.mockImplementation(()=>{
        return Promise.resolve([{
            "id": "e837c9f5-247f-445f-bcc3-7d434348336b",
            "firstName": "Martin",
            "lastName": "Bradley",
            "email": "mbradley0@google.it",
            "avatar": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
            "ip": "166.124.172.160"
          }]);
    });  

    it('renders snapshot as expected', ()=>{ 
        
        const wrapper = mount(<Provider store={store} > 
            <App/>
        </Provider>);

        return Promise.all([            
            messagePromise,
            memberPromise
        ]).then(()=>{   
            wrapper.update();            
            expect(toJson(wrapper, {noKey:true})).toMatchSnapshot();            
        });
    });
   
});