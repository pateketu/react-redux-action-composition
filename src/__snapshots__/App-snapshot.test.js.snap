// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot as expected 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App>
    <Connect(Chat)>
      <Chat
        chat={
          Array [
            Object {
              "id": "cd445e6d-e514-424f-ba8f-16ec842002c6",
              "message": "Test Message",
              "timestamp": "2017-02-09T04:27:38Z",
              "user": Object {
                "avatar": "http://dummyimage.com/100x100.png/5fa2dd/ffffff",
                "email": "mbradley0@google.it",
                "firstName": "Martin",
                "id": "e837c9f5-247f-445f-bcc3-7d434348336b",
                "ip": "166.124.172.160",
                "lastName": "Bradley",
              },
              "userId": "e837c9f5-247f-445f-bcc3-7d434348336b",
            },
          ]
        }
        loadChat={[Function]}
      >
        <div>
          <div
            className="chatItem"
          >
            Test Message
             by 
            <b>
              Martin
            </b>
              @  
            9/2/2017 4:27:38
          </div>
        </div>
      </Chat>
    </Connect(Chat)>
  </App>
</Provider>
`;
